packages/node-type-resolver/src/DependencyInjection/PHPStanServicesFactory.php

```diff
@@ -54,7 +56,11 @@ public function __construct()
 
             // bleeding edge clean out, see https://github.com/rectorphp/rector/issues/2431
             if (Strings::match($phpstanNeonContent, self::BLEEDING_EDGE_PATTERN)) {
-                $temporaryPhpstanNeon = $currentWorkingDirectory . '/rector-temp-phpstan.neon';
+                // Note: We need a unique file per process if rector runs in parallel
+                $pid = getmypid();
+                // and we use a temporary directory to not clutter the codebase in case
+                // the rector process is killed and doesn't clean up the files
+                $temporaryPhpstanNeon = $tempDirectory . '/rector-temp-phpstan' . $pid . '.neon';
```

Comment:
Using `$tempDirectory` here seems to break a lot of assumptions about phpstan includes that have been made in other places. Seems we're stuck with `$currentWorkingDirectory` for now :/

For reference:
```
1991) Rector\Core\Tests\Rector\Visibility\ChangePropertyVisibilityRector\ChangePropertyVisibilityRectorTest::test with data set #2 ('/codebase/rector/tests/Rector...hp.inc')
_HumbugBox09702017065e\Nette\FileNotFoundException: File '/tmp/utils/phpstan-extensions/config/phpstan-extensions.neon' is missing or is not readable.

Reply:
Could we handle this in standalone PR? I'd like to close the scope of this one first.

Reply:
Ah, I get it. It's a fix.